cocotb/__init__.py

```diff
@@ -129,11 +132,29 @@ def _reopen_stream_with_buffering(stream_name):
 """ used for cocotb library coverage """
 
 
-def fork(coro):
+def fork(coro: Union[RunningTask, coroutine]) -> RunningTask:
     """ Schedule a coroutine to be run concurrently. See :ref:`coroutines` for details on its use. """
     return scheduler.add(coro)
 
 
+def fork_soon(coro: Union[RunningTask, coroutine]) -> RunningTask:
+    """
+    Schedule a coroutine to be run concurrently, starting after the current coroutine yields control.
+
+    :func:`~cocotb.fork` starts the given coroutine immediately. This function
+    starts the given coroutine only after the current coroutine yields control.
+    This is useful when the forked coroutine has logic before the first
+    :keyword:`await` that may not be safe to execute immediately.
+    """
+    async def later():
+        await NullTrigger()
+        return await coro
+
+    scheduler.check_coroutine(coro)
+    update_wrapper(later, coro)
```

Comment:
I think this should only be used with coroutines.

Reply:
Why's that?

Reply:
I guess I meant it shouldn't be used for `RunningTask` objects. `@cocotb.coroutine` decorated functions are ok. If you pass in a `RunningTask`, `later` will end up waiting on `Join(<Task A>)`.

`update_wrapper` will change `later` to reflect the information on the Task A `RunningTask` object, so for Task B you'll have a repr of `<Task B coro=Task A() trigger=Join(<Task A>)>`. Scheduler debug will be wonky as well.

Reply:
I pushed some changes which should completely sidestep this problem by using `scheduler.queue` and not involving any intermediate coroutines at all.